allensdk/brain_observatory/behavior/behavior_project_cache/tables/util/metadata_parsers.py

```diff
@@ -0,0 +1,147 @@
+    "VisualBehaviorMultiscope": 2,
+    "VisualBehaviorMultiscope4areasx2d": 4,
+}
+
+NUM_DEPTHS_DICT = {
+    "VisualBehavior": 1,
+    "VisualBehaviorTask1B": 1,
+    "VisualBehaviorMultiscope": 4,
+    "VisualBehaviorMultiscope4areasx2d": 2,
+}
+
+
+############################
+#
+#     from session type
+#
+############################
+
+
+def parse_behavior_context(session_type: str) -> Optional[str]:
+    """Return if session is passive or active.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    behavior_type : str
+        Return string describing active or passive session.
+    """
+    is_passive = (
+        'passive' in session_type
+        and "OPHYS" in session_type
+    )
+    if is_passive:
+        behavior_type = "passive_viewing"
+    else:
+        behavior_type = "active_behavior"
+    return behavior_type
+
+
+def parse_stimulus_set(session_type: str) -> Optional[str]:
+    """Return the name of the image set or gratings.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    stim_set : str
+        Name of stimulus set shown for session.
+    """
+    session_type_split = session_type.split("_")
+    stimulus_type = session_type_split[2]
+    if stimulus_type == "images":
+        image_set_letter = session_type_split[3]
+        stim_set = f"{stimulus_type}_{image_set_letter}"
+    elif stimulus_type == "gratings":
+        stim_set = "gratings"
+    else:
+        raise ValueError(f"Session_type {session_type} not formatted as "
+                         "expected.")
+    return stim_set
+
+
+def add_experience_level(
+        sessions_df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Add the column 'experience_level' to a dataframe.
+
+    This column
+    will be 'Novel' for any rows with
+    'prior_exposures_to_image_set' == 0 (or NULL) and 'Familiar'
+    otherwise.
+
+    Parameters
+    ----------
+    sessions_df : pandas.DataFrame
+        Input sessions dataframe
+
+    Returns
+    -------
+    output_df : pandas.DataFrame
+        DataFrame with added "experience_level" column.
+    """
+    tmp_exposures = sessions_df['prior_exposures_to_image_set'].fillna(0)
+    sessions_df['experience_level'] = np.where(
+        tmp_exposures == 0,
+        'Novel',
+        'Familiar')
+    return sessions_df
+
+
+############################
+#
+#     from project code
+#
+############################
+
+
+def parse_num_cortical_structures(project_code: str) -> Optional[str]:
+    """Return the number of structures that were targeted for this session.
+
+    Parameters
+    ----------
+    project_code : str
+        Full project name of the experiment.
+
+    Returns
+    -------
+    num_structures : int
+        Number of structures targeted for the session.
+    """
+    return NUM_STRUCTURES_DICT.get(project_code, None)
```

Comment:
these are now really simple and aren't doing any parsing. Can you just directly use `NUM_STRUCTURES_DICT.get(project_code, None)` (same for `num_depths`) where it is being called rather than wrapping it in a function?

Reply:
I think having the function makes it a bit more clear what the task is rather than just having the `get` in pandas.apply.

Reply:
Ok but this isn't doing any parsing